src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/IllegalIdentifierNameCheck.java

```diff
@@ -0,0 +1,179 @@
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">
+ * ENUM_CONSTANT_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#PATTERN_VARIABLE_DEF">
+ * PATTERN_VARIABLE_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_DEF">
+ * RECORD_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_COMPONENT_DEF">
+ * RECORD_COMPONENT_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <p>Configuration:</p>
+ * <pre>
+ * &lt;module name="IllegalIdentifierName"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class TestClass {
+ *     public static void main(String... args) {
+ *         var var = 4; // violation, "var" should not be used as an identifier.
+ *         int record = 15; // violation, "record" should not be used as an identifier.
+ *         String yield = "yield"; // violation, "yield" should not be used as an identifier.
+ *
+ *         record Record // violation, "Record" should not be used as an identifier.
+ *             (Record record) { // violation, "record" should not be used as an identifier.
+ *         }
+ *
+ *         String yieldString = "yieldString"; // ok, part of another word
+ *         record MyRecord(){} // ok, part of another word
+ *         var variable = 2; // ok, part of another word
+ *         String _; // violation, underscore should not be used as an identifier.
+ *     }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to include "open" and "transitive" in the set of illegal identifiers:
+ * </p>
+ * <p>Configuration:</p>
+ * <pre>
+ * &lt;module name="IllegalIdentifierName"&gt;
+ *     &lt;property name="format" value="(?i)^(?!(record|yield|var|open|transitive|_)$).+$"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class TestClass {
+ *     public static void main(String... args) {
+ *
+ *         int open = 4; // violation, "open" should not be used as an identifier
+ *         Object transitive = "transitive"; // violation, "transitive" should not
+ *                                                // be used as an identifier
+ *
+ *         int openInt = 4; // ok, "open" is part of another word
+ *         Object transitiveObject = "transitiveObject"; // ok, "transitive" is part of another word
+ *     }
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code name.invalidPattern}
+ * </li>
+ * </ul>
+ *
+ * @since 8.36
+ */
+@StatelessCheck
+public class IllegalIdentifierNameCheck extends AbstractNameCheck {
+
+    /**
+     * Creates a new {@code IllegalIdentifierNameCheck} instance.
+     */
+    public IllegalIdentifierNameCheck() {
+        super("(?i)^(?!(record|yield|var|_)$).+$");
```

Comment:
After [JEP 360](https://openjdk.java.net/jeps/360) there will be more keywords in Java: `permits`, `sealed`, `non-sealed`.
The last one, `non-sealed` is not a valid identifier anyway, but `permits`, `sealed` should be in the list.

Reply:
Added.